packages/compass-connections/src/hooks/use-connection-status.ts

```diff
@@ -0,0 +1,44 @@
+import { useConnectionInfo } from '@mongodb-js/connection-storage/provider';
+import { useConnectionsManagerContext } from '../provider';
+import {
+  ConnectionStatus,
+  ConnectionsManagerEvents,
+} from '../connections-manager';
+
+import { type ConnectionInfo } from '@mongodb-js/connection-info';
+import { useEffect, useState } from 'react';
+
+export function useConnectionStatus(connectionInfo?: ConnectionInfo): {
+  status: ConnectionStatus;
+} {
+  if (!connectionInfo) {
+    connectionInfo = useConnectionInfo();
+  }
```

Comment:
It's not allowed to use hooks dynamically

Reply:
Ah right, I always forget about that rule.

I'll make the parameter mandatory.